src/MSubModuleChargeTransport.cxx

```diff
@@ -98,17 +148,23 @@ bool MSubModuleChargeTransport::AnalyzeEvent(MReadOutAssembly* Event)
   for (MDEEStripHit& SH: LVHits) {
     MVector Pos = SH.m_SimulatedPositionInDetector;
 
+    // Determine detector and strip dimensions from the geometry
+    unsigned int DetID = SH.m_ROE.GetDetectorID();
+    double XWidth = m_XWidths[DetID];
+    double YWidth = m_YWidths[DetID];
+    double XPitch = m_XPitches[DetID];
+    int NXStrips = m_NXStrips[DetID];
+
     // Calculate LV strip ID by rounding down intentionally to avoid truncation towards zero
-    // TODO: Confirm the correct strip pitch based on SMEX detector models
-    int ID = static_cast<int>(std::floor((Pos.X() + 7.4/2.0) / (7.4/64.0)));
+    int ID = static_cast<int>(std::floor((Pos.X() + XWidth/2.0) / XPitch));
 
     // Check for strip ID and if the position is within the allowed strip length or on the guard ring
     // TODO: Confirm the correct boundary of the guard ring based on SMEX detector models
-    if (ID >= 0 && ID < 64 && std::abs(Pos.Y()) <= 7.4/2.0 && std::hypot(Pos.X(), Pos.Y()) <= 4.712) {
+    if (ID >= 0 && ID < NXStrips && std::abs(Pos.Y()) <= YWidth/2.0 && std::hypot(Pos.X(), Pos.Y()) <= 4.712) {
```

Comment:
The only number that is not obvious for me to find is the radius of the guard ring for the edge strips, so that's why in here it's still `4.712`. Is that number stored somewhere in the geometry?

Reply:
I calculated the number from the strip length values that I saw in a Slack message. But I will look into getting them from the massmodel using the `WaferCutDisk` as suggested.